src/precision.hpp

```diff
@@ -0,0 +1,179 @@
+        {Precision::BF16, InferenceEngine::Precision::BF16},
+        {Precision::FP64, InferenceEngine::Precision::FP64},
+        {Precision::FP32, InferenceEngine::Precision::FP32},
+        {Precision::FP16, InferenceEngine::Precision::FP16},
+        {Precision::I64, InferenceEngine::Precision::I64},
+        {Precision::I32, InferenceEngine::Precision::I32},
+        {Precision::I16, InferenceEngine::Precision::I16},
+        {Precision::I8, InferenceEngine::Precision::I8},
+        {Precision::U64, InferenceEngine::Precision::U64},
+        {Precision::U16, InferenceEngine::Precision::U16},
+        {Precision::U8, InferenceEngine::Precision::U8},
+        {Precision::BOOL, InferenceEngine::Precision::BOOL},
+        {Precision::MIXED, InferenceEngine::Precision::MIXED},
+        {Precision::Q78, InferenceEngine::Precision::Q78},
+        {Precision::BIN, InferenceEngine::Precision::BIN},
+        {Precision::UNDEFINED, InferenceEngine::Precision::UNSPECIFIED},
+        {Precision::CUSTOM, InferenceEngine::Precision::CUSTOM}};
+    auto it = precisionMap.find(precision);
+    if (it == precisionMap.end()) {
+        return InferenceEngine::Precision::CUSTOM;  // TODO other way?
+    }
+    return it->second;
+}
+
+inline static ov::element::Type_t ovmsPrecisionToIE2Precision(Precision precision) {
+    static std::unordered_map<Precision, ov::element::Type_t> precisionMap{
+        {Precision::FP32, ov::element::Type_t::f32},
+        {Precision::FP16, ov::element::Type_t::f16},
+        {Precision::I64, ov::element::Type_t::i64},
+        {Precision::I32, ov::element::Type_t::i32},
+        {Precision::I16, ov::element::Type_t::i16},
+        {Precision::I8, ov::element::Type_t::i8},
+        {Precision::I4, ov::element::Type_t::i4},
+        {Precision::U64, ov::element::Type_t::u64},
+        {Precision::U16, ov::element::Type_t::u16},
+        {Precision::U8, ov::element::Type_t::u8},
+        {Precision::U4, ov::element::Type_t::u4},
+        {Precision::U1, ov::element::Type_t::u1},
+        {Precision::BOOL, ov::element::Type_t::boolean},
+        {Precision::UNDEFINED, ov::element::Type_t::undefined},  // TODO
+        {Precision::DYNAMIC, ov::element::Type_t::dynamic}       // TODO
+        //    {Precision::MIXED, ov::element::Type_t::MIXED}, // TODO
+        //    {Precision::Q78, ov::element::Type_t::Q78}, // TODO
+        //    {Precision::BIN, ov::element::Type_t::BIN}, // TODO
+        //    {Precision::CUSTOM, ov::element::Type_t::CUSTOM // TODO
+    };
+    auto it = precisionMap.find(precision);
+    if (it == precisionMap.end()) {
+        return ov::element::Type_t::undefined;  // TODO other way?
+    }
+    return it->second;
+}
+inline static Precision ovElementTypeToOvmsPrecision(ov::element::Type_t type) {
+    static std::unordered_map<ov::element::Type_t, Precision> precisionMap{
+        {ov::element::Type_t::f64, Precision::FP64},
```

Comment:
Type_t => Type

Reply:
there is no Type::f64. Type_t is enum while Type is wraparound class for that enum. There is converting constructor Type(Type_t). I am not sure at this point which we should use in the context of OVMS so I would refrain with potential swapping until the end of 2.0 conversion when we have full picture.